sonar-slang-plugin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -192,11 +174,60 @@ private Tree createElement(@Nullable PsiElement element) {
       return new LiteralTreeImpl(metaData, element.getText());
     } else if (element instanceof KtOperationExpression) {
       return createOperationExpression((KtOperationExpression) element, metaData);
+    } else if (element instanceof KtParameter) {
+      return new IdentifierTreeImpl(metaData, ((KtParameter) element).getName());
     } else {
       return new NativeTreeImpl(metaData, new KotlinNativeKind(element), list(Arrays.stream(element.getChildren())));
     }
   }
 
+  @NotNull
+  private Tree createFunctionDeclarationTree(@NotNull KtFunction functionElement, TreeMetaData metaData) {
+    List<Tree> modifiers = getModifierList(functionElement.getModifierList());
+    PsiElement nameIdentifier = functionElement.getNameIdentifier();
+    Tree returnType = null;
+    IdentifierTree identifierTree = null;
+    List<Tree> parametersList = list(functionElement.getValueParameters().stream());
+    Tree bodyTree = createElement(functionElement.getBodyExpression());
+    KtTypeElement typeElement = functionElement.getTypeReference() != null ? functionElement.getTypeReference().getTypeElement() : null;
+
+    if (functionElement.hasDeclaredReturnType() && typeElement != null) {
+      returnType = new IdentifierTreeImpl(getTreeMetaData(typeElement), typeElement.getText());
```

Comment:
do we need to test `functionElement.hasDeclaredReturnType()`?

Reply:
Nope it is indeed not necessary. Updated in bd329faf3c92cc56cc352e96e667a1ce9a5c06fa